src/actions/actions.ts

```diff
@@ -0,0 +1,66 @@
+import { Webhooks } from '../webhooks/webhooks';
+import { ResponsePayload } from './interfaces/response-payload';
+
+export class Actions {
+  private webhooks: Webhooks;
+
+  constructor(webhooks: Webhooks) {
+    this.webhooks = webhooks;
+  }
+
+  private get computeSignature() {
+    return this.webhooks.computeSignature.bind(this.webhooks);
+  }
+
+  get verifyHeader() {
+    return this.webhooks.verifyHeader.bind(this.webhooks);
+  }
+
+  async signResponse({
+    type,
+    verdict,
+    errorMessage,
```

Comment:
Alternative suggestion to this interface. Rather than 4 parameters, just pass two: all the payload data and the secret. The difference is `errorMessage` is not part of the type in the case of an `Allow`. It also leaves us open for the response interfaces to diverge per-action in the future. Thoughts?

It would look something like this:

```
async signResponse({
  responseData, // feel free to pick a better name here
  secret
}: {
  responseData: AuthenticationActionResponseData | UserRegistrationResponseData;
  secret: string;
}) ...

interface AllowResponseData {
  verdict: 'Allow';
}

interface DenyResponseData {
  verdict 'Deny';
  errorMessage?: string; // I assume we still want this to be optional, even in the event of a Deny
}

type AuthenticationActionResponseData = 
  AllowResponseData && { type: 'authentication' }
  | DenyResponseData && { type: 'authentication' }

type UserRegistrationResponseData = 
  AllowResponseData && { type: 'user_registration' }
  | DenyResponseData && { type: 'user_registration' }
```

Reply:
Great suggestion. Agree that gives us better flexibility here – it also matches the upstream API interfaces more closely.